### ISSUE TITLE
Plugin request for protoc-gen-go_{cli,gapic}

### ISSUE THREAD

**Issue by mfridman** [bufbuild/plugins#1305]
## Mandatory

**Where is the source code for the plugin?**

https://github.com/googleapis/gapic-generator-go/tree/main/cmd

## Optional

**Does the plugin have a valid semver version?**

Yes.

https://github.com/googleapis/gapic-generator-go/releases

Latest as of this writing is `v0.43.1`.

**Does the plugin have a dependency on another plugin?**

Not sure.

**Does the output of this plugin depend on any other external libraries?**

Not sure.

**Comment by bufdev** [comment]
Doesn't this plugin function by reading local disk if I remember? If so, we can't add it as of now, can you confirm @mfridman ?

**Comment by mfridman** [comment]
They made some significant changes in that project, I took a quick peek and most of the `os` calls are gated by an option. For example,

https://github.com/googleapis/gapic-generator-go/blob/134f862b5a05028c5e1af4b55ad213bf6ce9757d/internal/gengapic/generator.go#L137-L141

I'll timebox this and see if I can get a simple project to work without local disk.

TL; DR - Both plugins might work.
